google/resource_sql_database_instance.go

```diff
@@ -638,7 +647,17 @@ func resourceSqlDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
 	}
 
 	_settingsList := d.Get("settings").([]interface{})
-	_settings := _settingsList[0].(map[string]interface{})
+	var _settings map[string]interface{}
+
+	// If we're importing the settings will be nil
+	if len(_settingsList) > 0 {
+		_settings = _settingsList[0].(map[string]interface{})
+	} else {
+		_settingsList = make([]interface{}, 1)
```

Comment:
Yes, at the end of the Read, https://github.com/gh-mlfowler/terraform-provider-google/blob/245bf4359c166b683eda35b81881fab076cccc34/google/resource_sql_database_instance.go#L850-L851